### PR TITLE
🐛 Restrict range on Python version for `_make_forward_ref`

### PR DESCRIPTION
**Pull request by Kludex**
Reviewer, do you want to suggest a way to test this on the CI, or are we fine here?

Check https://github.com/pydantic/pydantic/discussions/6244#discussioncomment-6275458 for more information.

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by Kludex** [comment]
please review

**Comment by adriangb** [comment]
Please add a comment linking to the issue comment you linked to above. Otherwise LGTM, I don't think we need to test this in CI.

**Review by adriangb — APPROVED** [review]

File: pydantic/_internal/_typing_extra.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def _make_forward_ref(
         is_class: bool = False,
     ) -> typing.ForwardRef:
         """Wrapper for ForwardRef that accounts for the `is_class` argument missing in older versions.
-        The `module` argument is omitted as it breaks <3.9 and isn't used in the calls below.
+        The `module` argument is omitted as it breaks <3.9.8 and isn't used in the calls below.
 
         See https://github.com/python/cpython/pull/28560 for some background
```
Comment (Kludex) [inline pydantic/_internal/_typing_extra.py:265]:
```suggestion
        See https://github.com/python/cpython/pull/28560 for some background.
        The backport happened on 3.9.8, see: 
        https://github.com/pydantic/pydantic/discussions/6244#discussioncomment-6275458.
```

File: pydantic/_internal/_typing_extra.py [hunk 2]
```diff
@@ -260,9 +260,11 @@ def _make_forward_ref(
         is_class: bool = False,
     ) -> typing.ForwardRef:
         """Wrapper for ForwardRef that accounts for the `is_class` argument missing in older versions.
-        The `module` argument is omitted as it breaks <3.9 and isn't used in the calls below.
+        The `module` argument is omitted as it breaks <3.9.8 and isn't used in the calls below.
 
-        See https://github.com/python/cpython/pull/28560 for some background
+        See https://github.com/python/cpython/pull/28560 for some background.
+        The backport happened on 3.9.8, see:
```
Comment (Kludex) [inline pydantic/_internal/_typing_extra.py:266]:
```suggestion
        The backport happened on 3.9.8, see:
```

**Pull request merged**